virtool/blast/task.py

```diff
@@ -41,57 +41,96 @@ def __init__(
 
         self.analysis_id = self.context["analysis_id"]
         self.sequence_index = self.context["sequence_index"]
-        self.interval = 3
-        self.steps = [self.request, self.wait]
+        self.interval: Optional[int] = None
+        self.steps = [self.request]
         self.rid: Optional[str] = None
 
     async def request(self):
-        """Make the initial request to NCBI to start a BLAST search."""
-        blast = await self.data.blast.initialize_on_ncbi(
-            self.analysis_id, self.sequence_index
-        )
+        """
+        Make the initial request to NCBI to start a BLAST search.
 
-        self.rid = blast.rid
+        Checks are conducted by the data layer and will store the results or error
+        when the search completes. The task completes when either an error or
 
-    async def wait(self):
+        The BLAST will be retried up to 3 times if a single BLAST
+        search exceeds 10 minutes.
         """
-        Wait until the BLAST search completes.
 
-        Keep check the BLAST status on NCBI with increasingly longer intervals between
-        checks.
+        blast_timeout_count: int = 0
 
-        Checks are conducted by the data layer and will store the results or error when
-        the search completes. The task completes when either an error or
+        while True:
```

Comment:
I don't think your `while...break` pattern is doing anything. Look closely and either fix or reply back with explanation.

Another point, if an exception is raised in your `while` block, it will break the loop. You should put the `try` outside the loop and handle exceptions outside the loop as well.